mlflow/models/signature.py

```diff
@@ -51,23 +51,30 @@
 
 class ModelSignature:
     """
-    ModelSignature specifies schema of model's inputs and outputs.
+    ModelSignature specifies schema of model's inputs, outputs and params.
 
-    ModelSignature can be :py:func:`inferred <mlflow.models.infer_signature>` from training dataset
-    and model predictions using or constructed by hand by passing an input and output
-    :py:class:`Schema <mlflow.types.Schema>`.
+    ModelSignature can be :py:func:`inferred <mlflow.models.infer_signature>` from training
+    dataset, model predictions using and params for inference, or constructed by hand by
+    passing an input and output :py:class:`Schema <mlflow.types.Schema>`, and params
+    :py:class:`ParamSchema <mlflow.types.ParamSchema>`.
     """
 
-    def __init__(self, inputs: Schema, outputs: Schema = None):
+    def __init__(self, inputs: Schema, outputs: Schema = None, params: ParamSchema = None):
         if not isinstance(inputs, Schema):
             raise TypeError(f"inputs must be mlflow.models.signature.Schema, got '{type(inputs)}'")
         if outputs is not None and not isinstance(outputs, Schema):
             raise TypeError(
                 "outputs must be either None or mlflow.models.signature.Schema, "
                 "got '{}'".format(type(inputs))
             )
+        if params and not isinstance(params, ParamSchema):
+            raise TypeError(
+                "params must be either None or mlflow.models.signature.ParamSchema, "
+                "got '{}'".format(type(params))
```

Comment:
silly nit:

```suggestion
                "if params are provided, they must be of type mlflow.models.signature.ParamSchema. "
                "got '{}'".format(type(params))
```
I know this mirrors the TypeError from above, but that one also is a little confusing to a user :)